application/common/models/EmailLog.php

```diff
@@ -118,6 +118,41 @@ public function rules()
                 'default',
                 'value' => MySqlDateTime::now(),
             ],
+            [
+                'non_user_address',
+                'emptyIfUserIsSet',
+            ],
+            [
+                'non_user_address',
+                'requiredIfUserIsEmpty',
+                'skipOnEmpty' => false,
+            ],
         ], parent::rules());
     }
+
+    public function emptyIfUserIsSet($attribute)
```

Comment:
These validators don't read well as standalone functions. At least it took me more time to figure them out for as simple as they should be. I couldn't think of a better name for the function so I just wrote a function header.
```suggestion

    /**
     * Validate an attribute to ensure it is empty (unset) if `user_id` is set.
     *
     * @param string $attribute Attribute name to be validated
     * @return bool
     */
    public function emptyIfUserIsSet($attribute)
```

Reply:
Actually, using a validator class for each function is a better implementation, in my opinion. Individual functions like this are so Yii1.

Reply:
Thanks, @briskt.

And you're right, @mtompset: a validator class is probably a better way to do this. I opted for the in-class validator function for these, since it was a smaller step to take. We'll try proceeding with the DocBlock that @briskt added, but if these getting any bigger, moving them to validator classes is a good idea.